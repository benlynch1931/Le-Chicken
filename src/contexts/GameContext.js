import React, { createContext, Component } from 'react';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

export const GameContext = createContext();

const chickenPositionDefault = [wp("43.34%"), hp("60%")];
class GameContextProvider extends Component {
  state = {
    currentScene: 'coop',
    chickenGraphic: 'idleright',
    chickenPosition: chickenPositionDefault,
    inputText: "",
    hint: "Pour marcher: Type ‘marcher’",
    chickenToMove: 0,
    level: 0,
    chickenDirection: "up",
    walls: [{
      type: 'horizontal',
      position: 48.34,
      start: 56.67,
      end: 100.00,
      stroke: 1.85
    }, {
      type: 'vertical',
      position: 2.00,
      start: 0.00,
      end: 49.26,
      stroke: 4.00
    }]
  }



  changeLevel = (level) => {
    this.setState({ level: level })
  }
  changeChickenDirection = (direction) => {
    this.setState({ chickenDirection: direction })
  }
  changeInputText = (inputText) => {
    this.setState({ inputText: inputText })
  }
  changeScene = (sceneName) => {
    this.setState({ currentScene: sceneName })
  }
  changeChickenGraphic = (chickenGraphic) => {
    this.setState({ chickenGraphic: chickenGraphic })
  }
  increaseChickenPosition = (x, y) => {
    this.setState({ chickenPosition: [this.state.chickenPosition[0] + x, this.state.chickenPosition[1] + y] })
  }
  resetChickenPosition = () => {
    this.setState({ chickenPosition: chickenPositionDefault })
    this.changeChickenGraphic('up')
  }

  changeHint = (hint) => {
    this.setState({ hint: hint })
  }

  changeChickenToMove = (chickenToMove) => {
    this.setState({ chickenToMove: chickenToMove })
  }

  render() {
    return (
      <GameContext.Provider value={{
        ...this.state,
        changeScene: this.changeScene,
        changeChickenGraphic: this.changeChickenGraphic,
        increaseChickenPosition: this.increaseChickenPosition,
        resetChickenPosition: this.resetChickenPosition,
        changeInputText: this.changeInputText,
        changeHint: this.changeHint,
        changeChickenToMove: this.changeChickenToMove,
        changeChickenDirection: this.changeChickenDirection,
        changeLevel: this.changeLevel
      }}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}

export default GameContextProvider;
