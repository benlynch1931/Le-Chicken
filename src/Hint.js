import React, { useContext, useEffect, useState } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { GameContext } from './contexts/GameContext.js';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';


const Hint = () => {
  const { level } = useContext(GameContext)
  const [currentHint, setCurrentHint] = useState("Pour marcher: Type ‘marcher’")
  const hints = {
    0: "Pour marcher: Type ‘marcher’",
    1: "Pour ouvrir: Type 'ouvrir'",
    2: "Check your dictionary for [HAUT], [DROITE], [BAS], [GAUCHE]"
  }

  useEffect(() => {
    setCurrentHint(hints[level])
  }, [level, currentHint])

  return (
    < View style={styles.container} >
      <Text style={styles.hintText}>Hint: {currentHint}</Text>
    </View >
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    width: wp('100%'),
  },
  hintText: {
    alignSelf: "center",
    padding: hp("0.8%"),
    fontSize: 10
  }
});
export default Hint;
