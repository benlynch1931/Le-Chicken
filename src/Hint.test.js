import React from 'react';
import { create, act } from 'react-test-renderer';

import Hint from './Hint.js';
import { GameContext } from './contexts/GameContext.js';

const mockContext = {level: 0}

test('displays correct hint at level 0', () => {
    const hint = create(<GameContext.Provider value={mockContext}><Hint/></GameContext.Provider>)
    expect(hint).toMatchSnapshot();
})

test('displays correct hint at level 1', async () => {
    mockContext.level = 1
    await act(async () => {
        newHint = create(<GameContext.Provider value={mockContext}><Hint/></GameContext.Provider>)
    })
    expect(newHint).toMatchSnapshot();
})
