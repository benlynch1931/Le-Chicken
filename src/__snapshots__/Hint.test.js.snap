// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct hint at level 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "width": 750,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "Pixel",
        "fontSize": 10,
        "lineHeight": 18,
        "padding": 10.5,
      }
    }
  >
    Pour marcher: Type ‘marcher’
  </Text>
</View>
`;

exports[`displays correct hint at level 1 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "width": 750,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "Pixel",
        "fontSize": 10,
        "lineHeight": 18,
        "padding": 10.5,
      }
    }
  >
    Hint: Pour ouvrir: Type 'ouvrir'
  </Text>
</View>
`;
